Pulled from upstream.
https://github.com/antirez/redis/commit/0c211a1953afeda3d0d45126653e2d4c38bd88cb
"Simplify lua_cmsgpack macro and fix build on old Linux distros."

diff -Naur a/deps/lua/src/lua_cmsgpack.c b/deps/lua/src/lua_cmsgpack.c
--- a/deps/lua/src/lua_cmsgpack.c	2014-12-04 04:36:52.000000000 -0600
+++ b/deps/lua/src/lua_cmsgpack.c	2014-12-05 10:28:23.827575434 -0600
@@ -18,14 +18,8 @@
     #define LUACMSGPACK_MAX_NESTING  16 /* Max tables nesting. */
 #endif
 
-#if (_XOPEN_SOURCE >= 600 || _ISOC99_SOURCE || _POSIX_C_SOURCE >= 200112L)
-    #define IS_FINITE(x) isfinite(x)
-#else
-    #define IS_FINITE(x) ((x) == (x) && (x) + 1 > (x))
-#endif
-
 /* Check if float or double can be an integer without loss of precision */
-#define IS_INT_TYPE_EQUIVALENT(x, T) (IS_FINITE(x) && (T)(x) == (x))
+#define IS_INT_TYPE_EQUIVALENT(x, T) (!isinf(x) && (T)(x) == (x))
 
 #define IS_INT64_EQUIVALENT(x) IS_INT_TYPE_EQUIVALENT(x, int64_t)
 #define IS_INT_EQUIVALENT(x) IS_INT_TYPE_EQUIVALENT(x, int)
